### DISCUSSION TITLE
Square Axis for 3D plots

### DISCUSSION THREAD

**Discussion by acxz** [alandefreitas/matplotplusplus#99] [Ideas]
I've been working with 3D plots lately and it would be pretty nice to have an option of `square3` similar to the 2D option of `square` as done here: https://github.com/alandefreitas/matplotplusplus/blob/52d9e75f62474230cd365250d1484c43d3de55d1/source/matplot/freestanding/axes_functions.cpp#L625

It seems it wouldn't be too hard to add, but I'm not sure if we would need an `equal3` option as well.

Looking closer it seems so:
https://github.com/alandefreitas/matplotplusplus/blob/52d9e75f62474230cd365250d1484c43d3de55d1/source/matplot/freestanding/axes_functions.cpp#L574-L582

Any thoughts?

**Comment by alandefreitas** [comment]
Sounds like a good idea. We can just use the same `square` option and make this function behave a little differently if `ax->is_3d()`.
